src/wallet/wallet.cpp

```diff
@@ -2820,6 +2818,12 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     }
                 }
 
+                // If subtracting fee from recipients, we now know what fee we
+                // need to subtract, we have no reason to reselect inputs
+                if (nSubtractFeeFromAmount && nFeeRet == 0) {
```

Comment:
`nFeeRet == 0` should not be necessary? So the first iteration should have enough inputs for when `nSubtractFeeFromAmount > 0`, or I am missing something?